bundle/config/mutator/resolve_variable_references.go

```diff
@@ -205,3 +227,45 @@ func (m *resolveVariableReferences) resolveOnce(b *bundle.Bundle, prefixes []dyn
 
 	return hasUpdates, diags
 }
+
+// selectivelyMutate applies a function to a subset of the configuration
+func (m *resolveVariableReferences) selectivelyMutate(b *bundle.Bundle, fn func(value dyn.Value) (dyn.Value, error)) error {
+	included := []string{
+		"variables",
+		"bundle",
+		"include",
+		"workspace",
+		"artifacts",
+		"targets",
+		"environments",
+		"sync",
+		"run_as",
+		"presets",
+		"experimental",
+		"permissions",
```

Comment:
Wouldn't it be more future-proof to take root dyn value, convert to map and read keys from it?

Reply:
Pieters point was "+1 that would make this simpler to read. And it forces listing the top level keys whereas it is implicit now, so seems like a win-win."

If we iterate root dyn value it will be implicit.

Reply:
I changed it as you suggested. It doesn't look as readable as in Python, because we program golang